Gems/Terrain/Assets/Shaders/Terrain/ClipmapComputeHelpers.azsli

```diff
@@ -188,41 +188,90 @@ struct ClipmapLevel
+    // Calculate the blending fact if the position falls into the blending band.
+    float2 clipmapCenterInWorldSpace = GetMacroClipmapCenterInWorldSpace(clipmapLevel.m_closestLevel);
+    float clipmapToWorldScale = GetMacroClipmapToWorldScale(clipmapLevel.m_closestLevel);
 
-    float renderDistance = maxRenderDistance / pow(clipmapScaleBase, (float)(clipmapStackSize - 1u - clipmapLevel.m_closestLevel));
-    float blendingStart = renderDistance * BlendingStartFactor;
-    float blendingEnd = renderDistance / RenderDistanceBias;
-    float blendingValue = max(absDistance.x, absDistance.y);
+    float blendBeginDistance = clipmapSize / 2.0 - clipmapMarginSize - ExtendedClipmapMarginSize - BetweenLevelsBlendBandSize;
+    // Clipmap uses Chebyshev distance
+    float2 distance = abs(worldPosition - clipmapCenterInWorldSpace);
+    float cDistance = max(distance.x, distance.y) / clipmapToWorldScale;
+    clipmapLevel.m_weight = max(0.0, (cDistance - blendBeginDistance) / BetweenLevelsBlendBandSize);
 
-    if (blendingValue > blendingStart && blendingValue < blendingEnd)
-    {
-        clipmapLevel.m_weight = (blendingValue - blendingStart) / (blendingEnd - blendingStart);
-        clipmapLevel.m_nextLevel = clipmapLevel.m_closestLevel + 1;
-    }
-    else
+    return clipmapLevel;
+}
+
+ClipmapLevel CalculateDetailClipmapLevel(float2 worldPosition)
+{
+    uint clipmapSize = TerrainSrg::m_clipmapData.m_clipmapSizeUint;
+    uint clipmapMarginSize = TerrainSrg::m_clipmapData.m_detailClipmapMarginSize;
+
+    ClipmapLevel clipmapLevel;
+    float closestLevel = 1.0;
+    // ignore the last level, as it will be used as a fall back default.
+    uint clipmapStackSize = TerrainSrg::m_clipmapData.m_detailClipmapStackSize - 1;
+    for (uint clipmapIndex = 0; clipmapIndex < clipmapStackSize; ++clipmapIndex)
     {
-        clipmapLevel.m_weight = 0.0;
-        clipmapLevel.m_nextLevel = clipmapStackSize;
+        float2 clipmapCenterInWorldSpace = GetDetailClipmapCenterInWorldSpace(clipmapIndex);
+        float clipmapToWorldScale = GetDetailClipmapToWorldScale(clipmapIndex);
+        float extent = (clipmapSize / 2.0 - clipmapMarginSize - ExtendedClipmapMarginSize) * clipmapToWorldScale;
+
+        float contains = AabbContainsPoint(clipmapCenterInWorldSpace, extent, worldPosition);
+        closestLevel += contains;
     }
+    clipmapLevel.m_closestLevel = TerrainSrg::m_clipmapData.m_detailClipmapStackSize - uint(closestLevel);
+    clipmapLevel.m_nextLevel = clipmapLevel.m_closestLevel + 1;
+
+    // Calculate the blending factor if the position falls into the blending band.
+    float2 clipmapCenterInWorldSpace = GetDetailClipmapCenterInWorldSpace(clipmapLevel.m_closestLevel);
+    float clipmapToWorldScale = GetDetailClipmapToWorldScale(clipmapLevel.m_closestLevel);
+
+    float blendBeginDistance = clipmapSize / 2.0 - clipmapMarginSize - ExtendedClipmapMarginSize - BetweenLevelsBlendBandSize;
```

Comment:
it look like every time clipmapSize and clipmapMarginSize are used, it's in the equation 
```
clipmapSize / 2.0 - clipmapMarginSize - ExtendedClipmapMarginSize
```
This could be calculated on the cpu side and sent in as one value.